File: src/Java/Behavioral/Strategy/DispatchStrategy/PrivateMessageStrategy.java
package Java.Behavioral.Strategy.DispatchStrategy;

import Java.Behavioral.Strategy.Message;

public class PrivateMessageStrategy implements DispatchStrategy {
    @Override
    public void send(Message message) {
        System.out.println("Message sent to " + message.getReceivingUserDetails().getFullName() + " via private message");
        System.out.println("Content: " + message.getContent());
        System.out.println("Send Timestamp: " + message.getDateTimeSent() + "\n");
    }
}

File: src/Kotlin/strategyPattern/DispatchStrategies/PrivateMessageStrategy.kt
package Kotlin.strategyPattern.DispatchStrategies

import Kotlin.strategyPattern.Message

class PrivateMessageStrategy : DispatchStrategy {

    override fun send(message: Message) {
        println("Message sent to " + message.receivingUser.fullName + "Via private message")
        println("Content: " + message.content)
        println("Send Timestamp: " + message.dateTimeSent + "\n")
    }
}
